cupy/cuda/stream.pyx

```diff
@@ -411,14 +416,13 @@ class ExternalStream(BaseStream):
     """
 
     def __init__(self, ptr, device_id=-1):
-        self.ptr = ptr
         # It is in theory unsafe to just call runtime.getDevice() here, as the
         # stream pointer could come from a different device (although
         # unlikely). While we could use driver API combos cuStreamGetCtx ->
         # cuCtxSetCurrent -> cuCtxGetDevice -> ... to retrieve the device ID
-        # associated with the stream, it is way too complicated. Let us keep
-        # this as thin as possible.
-        self.device_id = device_id
+        # associated with the stream, it is way too complicated and does not
+        # work with HIP. Let us keep this as thin as possible.
```

Comment:
This is referring to https://github.com/cupy/cupy/pull/5172#discussion_r625841360. Thanks for adding it!